### PR TITLE
Introduce `pyproject.toml` + pdm

### LINKED ISSUE #345: Add `pyproject.toml` project metadata file backended with building/packaging/dependency manager.

### PR DESCRIPTION
**Pull request by stankudrow**
Fixes #345 .

A variation of #346 .

Checklist:

- [ ] Add tests that demonstrate the correct behaviour of the change. Tests should fail without the change. -> NO NEED TO.
- [ ] Add or update relevant docs, in the docs folder and in code. -> NO NEED TO
- [x] Ensure PR doesn't contain untouched code reformatting: spaces, etc.
- [x] Run `flake8` and fix issues. -> WITH RUFF
- [x] Run `pytest` no tests failed. See https://clickhouse-sqlalchemy.readthedocs.io/en/latest/development.html. -> DONE UNDER PY3.9

### CODE REVIEW

**Comment by stankudrow** [comment]
@xzkostyan , @DaniilAnichin, let's give this PR a try in favour of #346 .

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,96 @@
+[project]
+name = "clickhouse-sqlalchemy"
+dynamic = ["version"]
+description = "SQLAlchemy dialect for ClickHouse"
+authors = [
+    {name = "Konstantin Lebedev", email = "kostyan.lebedev@gmail.com"},
+]
+dependencies = ["alembic>=1.14.0", "asynch>=0.2.5", "clickhouse-driver>=0.2.9", "requests>=2.32.3", "sqlalchemy>=2.0.36"]
```
Comment (dolfinus) [inline pyproject.toml:8]: Is alembic dependency really required by this package?

File: testsrequire.py [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
-    'responses',
-    'parameterized'
+    "pytest",
+    "pytest-asyncio",
+    "sqlalchemy>=2.0.0,<2.1.0",
+    "greenlet>=2.0.1",
+    "alembic",
+    "requests",
+    "responses",
+    "parameterized",
```
Comment (stankudrow) [inline testsrequire.py:9]: Maybe getting rid of this file and just launching `pytest`?

File: tests/types/test_enum16.py [hunk 3]
```diff
@@ -8,7 +8,7 @@
 from tests.util import with_native_and_http_sessions
 
 
-class TestEnum(enum.IntEnum):
+class _TestEnum(enum.IntEnum):
```
Comment (stankudrow) [inline tests/types/test_enum16.py:11]: https://github.com/xzkostyan/clickhouse-sqlalchemy/pull/348/files#r1835443711

File: pyproject.toml (hunk 1 above)
Comment (stankudrow) [reply pyproject.toml]: Thank you, you are right, alembic got moved into the "test" dependency group.

**Comment by dolfinus** [comment]
Sorry, but I'm not related to this project at all, and have no permissions to approve this PR. Could you please remove me from reviewers?

**Comment by stankudrow** [comment]
> Sorry, but I'm not related to this project at all, and have no permissions to approve this PR. Could you please remove me from reviewers?

Hm, I cannot or I don't see the option.

**Comment by stankudrow** [comment]
@xzkostyan , hello, sorry for pinging. Could you assign a reviewer for this PR and name some maintainers to tag for not disturbing you?

**Comment by stankudrow** [comment]
@xzkostyan , hello. Could you possibly have a look at this PR?

**Comment by stankudrow** [comment]
What is moving this project to [uv](https://docs.astral.sh/uv/) 🤔 .

**Pull request closed without merging**